python/MooseDocs/extensions/MooseImageFile.py

```diff
@@ -0,0 +1,103 @@
+import re
+import os
+from markdown.util import etree
+import logging
+log = logging.getLogger(__name__)
+
+import MooseDocs
+from markdown.inlinepatterns import Pattern
+from MooseCommonExtension import MooseCommonExtension
+import utils
+
+class MooseImageFile(MooseCommonExtension, Pattern):
+    """
+    Markdown extension for handling images.
+    """
+
+    # Find !image /path/to/file attribute=setting
+    RE = r'!image\s+(.*?)\s+(.*)'
+    SETTINGS_RE = re.compile("([^\s=]+)=(.*?)(?=(?:\s[^\s=]+=|$))")
+
+    def __init__(self, media_dir=None, root=None, **kwargs):
+        MooseCommonExtension.__init__(self)
+        Pattern.__init__(self, self.RE, **kwargs)
+
+        self._root = os.path.join(root, media_dir)
+
+        # The default settings
+        self._settings = {'alt'     : None,
+                          'caption' : None,
+                          'width'   : None,
+                          'height'  : None,
+                          'align'   : 'left'}
+
+
+
+    def handleMatch(self, match):
+        """
+        Returns a tree element containing error message.
+
+        Uses the html to match the python markdown admonition package.
+        https://pythonhosted.org/Markdown/extensions/admonition.html
+
+        <div class="admonition danger">
+        <p class="admonition-title">Don't try this at home</p>
+        <p>...</p>
+        </div>
+        """
+
+        rel_filename = match.group(2)
+
+        # Perform second regex to capture key=value pairs with
+        # possible spaces in the value
+        settings = self.parseOptions(match.group(3))
+
+        # Read the file and create element
+        filename = self.checkFilename(rel_filename)
+
+        # caption and alt are one and the same
+        if settings['caption'] != None:
+            settings['alt'] = settings['caption']
+
+        if not filename:
+            print '{}'.format(rel_filename)
+            el = self.createErrorElement(rel_filename, message='file not found')
+
+        elif settings.has_key('table'):
+            # TODO
+            # populate table options based on settings
+            table_options = {}
+            table_dict = self.createTableElement(rows=1, columns=1, **table_options)
+
+            el = table_dict['table']
+            img = etree.SubElement(table_dict['td'][0], 'img')
+            img.set('src', os.path.join('../../media', rel_filename))
+
+            # if any attributes for img, set them now
+            for attribute in settings.keys():
+                if settings[attribute] != None:
+                    img.set(attribute, str(settings[attribute]))
+
+            if settings['alt'] != None:
+                table_dict['td'][0].text = settings['alt']
```

Comment:
As we discussed, I don't think we will need this. I really would rather see markdown tables.